ouroboros-network/default.nix

```diff
@@ -0,0 +1,38 @@
+{ mkDerivation, aeson, array, async, base, base16-bytestring
+, bytestring, cborg, clock, containers, cryptonite, directory
+, exceptions, filepath, fingertree, free, hashable, memory, mtl
+, network, optparse-applicative, pipes, process, psqueues
+, QuickCheck, random, semigroups, serialise, stdenv, stm
+, string-conv, tasty, tasty-expected-failure, tasty-hunit
+, tasty-quickcheck, temporary, text, time, transformers
+, typed-transitions, unix, unliftio, vector, void
+}:
+mkDerivation {
+  pname = "ouroboros-network";
+  version = "0.1.0.0";
+  src = ./.;
```

Comment:
It's better to use:
```
  src = nixpkgs.lib.sourceFilesBySuffices ./.
    [ ".hs" "LICENSE" "ChangeLog.md" "ouroboros-network.cabal" "cabal.project" ];
```
This way one does not put everything inside nix build environment, e.g. a tags file etc...

Reply:
Would be better to have a blacklist rather than a whitelist.

Reply:
Or maybe not... Is this `nixpkgs.lib.sourceFilesBySuffixes` considered standard practice? The files I put in here are generated by `cabal2nix`.

Reply:
I got once into trouble because some tool was putting files into project directory, and then I was wondering why I every time things are rebuild.  Somebody gave me advice to use this snippet.  The `nixpkgs.lib` is pretty small, and this is the easiest solution that comes with `nixpkgs`.  White listing avoids this, and it worked fine for me in various other projects (though much smaller).

Another modification I tend to use is `  enableSeparateDocOutput = false;`  Then the docs are included in `./results`.  It makes it easier to access them.

I will make a PR and ask somebody from devops to give us an advice.